Source/PINDiskCache.m

```diff
@@ -838,7 +838,7 @@ - (id)objectForKeyedSubscript:(NSString *)key
 {
     NSDate *now = [[NSDate alloc] init];
     
-    if (!key)
+    if (!key || (_dates.count == 0 && _sizes.count == 0))
```

Comment:
You need a lock here both because this is not thread-safe (at least not formally) and because the disk cache may not have been initialized yet.